Yoast/wordpress-seo
# P2-465 Only restore the indexing notification when the indexing reason has changed.

## Context


* We only want to restore (e.g. unhide) the indexing notification when the indexing reason has changed. Not on every page reload.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the indexing notification was always shown, even when it was hidden before.

## Relevant technical choices:

* ~I decided to store the indexing reason on the notification, so we can compare the old reason and the new reason.~
* The notification is now removed (and re-added again) whenever the indexing reason changes.
  * When 'Create indexing helper to get and set options' (#16222) is merged this removal can be added to the `set_reason` method in the `indexing-helper` class. In this PR, since it is based on the release branch, I had to remove the notification wherever the reason is set, unfortunately.
* I added a check in the notification integration to check whether the notification is already in the notification center. If not, and it should be shown, it is re-added and restored (unhidden).
  * This makes sure that it is re-added when the indexing reason changes.

## Test instructions

This PR can be tested by following these steps:

**Please note**: this PR [has an accompanying PR in Premium](https://github.com/Yoast/wordpress-seo-premium/pull/3052).

* Change the category base of your permalinks in your WP admin settings.
  * In _Settings_ > _Permalinks_.
* Go to your Yoast SEO dashboard.
* You should see a notification indicating that your site needs reindexing since the permalink settings have been changed.
* Hide the notification using the eye icon on the right of the notification.
* Reload the page.
* The notification should still be hidden.
* Change your permalink structure in your WP admin settings.
  * In _Settings_ > _Permalinks_.
* Go to your Yoast SEO dashboard.
* You should see a notification indicating that your site needs reindexing since the category base has been changed.
* This notification should **not** be hidden anymore.

## UI changes

## Documentation

## Quality assurance


Fixes #
